RestClientBuilderImpl.verifyClient throws an exception when Path annotation value on interface is empty

When trying to create a new rest client based on an interface like the one below,
with an empty @Path annotation value,

```
@Path("")
public interface RestClientInterfaceWithEmptyPath {
    @Path("/{aPathParameter}")
    @GET
    String get(@PathParam("aPathParameter") String aPathParameter);
}
```

```
RestClientBuilder.newBuilder()
        .baseUri("https://www.example.com")
        .build(RestClientInterfaceWithEmptyPath.class);
```

you get an error:

```
[ERROR] Errors: 
[ERROR]   RestClientInterfaceWithEmptyPathTest.testRestClientInterface:11 » RestClientDefinition Parameters and variables don't match on interface com.acme.RestClientInterfaceWithEmptyPath::get
```

This is caused by the method ResteasyUriBuilderImpl.uri interpreting the first parameter in the path as the host.

Using the method ResteasyUriBuilderImpl.path fixes the issue.

The Quarkus REST client builder (based on this) had the same issue and it was resolved in this way
(see https://github.com/quarkusio/quarkus/pull/33918).

You can find a reproducer here:
https://github.com/jacopotessera/resteasy-microprofile-verifyinterface

I'm opening a PR with this fix

--- a/rest-client-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java
+++ b/rest-client-base/src/main/java/org/jboss/resteasy/microprofile/client/RestClientBuilderImpl.java
@@ -606,9 +606,9 @@ public class RestClientBuilderImpl implements RestClientBuilder {
             if (methodPathAnno != null) {
                 template = classPathAnno == null ? (ResteasyUriBuilder) new ResteasyUriBuilderImpl().uri(methodPathAnno.value())
                         : (ResteasyUriBuilder) new ResteasyUriBuilderImpl()
-                                .uri(classPathAnno.value() + "/" + methodPathAnno.value());
+                                .path(classPathAnno.value() + "/" + methodPathAnno.value());
             } else if (classPathAnno != null) {
-                template = (ResteasyUriBuilder) new ResteasyUriBuilderImpl().uri(classPathAnno.value());
+                template = (ResteasyUriBuilder) new ResteasyUriBuilderImpl().path(classPathAnno.value());
             } else {
                 template = null;
             }

--- a/rest-client/src/test/java/org/jboss/resteasy/microprofile/client/RestClientWithEmptyPath.java
+++ b/rest-client/src/test/java/org/jboss/resteasy/microprofile/client/RestClientWithEmptyPath.java
@@ -1,0 +1,12 @@
+package org.jboss.resteasy.microprofile.client;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+
+@Path("")
+public interface RestClientWithEmptyPath {
+    @Path("/{aPathParameter}")
+    @GET
+    String get(@PathParam("aPathParameter") String aPathParameter);
+}

--- a/rest-client/src/test/java/org/jboss/resteasy/microprofile/client/RestClientWithEmptyPathTest.java
+++ b/rest-client/src/test/java/org/jboss/resteasy/microprofile/client/RestClientWithEmptyPathTest.java
@@ -1,0 +1,15 @@
+package org.jboss.resteasy.microprofile.client;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class RestClientWithEmptyPathTest {
+    @Test
+    public void testBuildRestClient() {
+        // the client get created without errors
+        Assertions.assertDoesNotThrow(() -> RestClientBuilder.newBuilder()
+                .baseUri("http://localhost")
+                .build(RestClientWithEmptyPath.class));
+    }
+}

